### PR TITLE
MDEV-19160 JSON_DETAILED output unnecessarily verbose

### PR DESCRIPTION
**Pull request by Weijun-H**
- [x] *The Jira issue number for this PR is: MDEV-19160*

## Description
Represents JSON in the most understandable way emphasizing nested structures.
## How can this PR be tested?
The test files locate at `func_json_detailed.test` and `func_json_detailed.result`

## Basing the PR against the correct MariaDB version
- [x] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [ ] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

### CODE REVIEW

File: sql/item_jsonfunc.cc [hunk 1]
```diff
@@ -377,18 +389,34 @@ static int json_nice(json_engine_t *je, String *nice_js,
...
       first_value= 0;
+      value_size= -1;
       break;
 
     default:
       break;
     };
   } while (json_scan_next(je) == 0);
-
+
```
Comment (mariadb-RuchaDeodhar) [inline sql/item_jsonfunc.cc:419]: Please remove this empty space.

File: sql/item_create.cc [hunk 2]
```diff
@@ -5762,6 +5762,7 @@ Native_func_registry func_array[] =
   { { STRING_WITH_LEN("JSON_CONTAINS_PATH") }, BUILDER(Create_func_json_contains_path)},
   { { STRING_WITH_LEN("JSON_DEPTH") }, BUILDER(Create_func_json_depth)},
   { { STRING_WITH_LEN("JSON_DETAILED") }, BUILDER(Create_func_json_detailed)},
+  { { STRING_WITH_LEN("JSON_PRETTY") }, BUILDER(Create_func_json_detailed)},
```
Comment (mariadb-RuchaDeodhar) [inline sql/item_create.cc:5765]: This looks ok. But could you make a separate patch for json_pretty() ?

File: mysql-test/main/func_json_detail.test [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        "steps": []
+      }
+    }
+  ]
+}');
+
+
+select JSON_DETAILED(JSON_EXTRACT(a, '$**.analyzing_range_alternatives')) from t200;
+select JSON_PRETTY(JSON_EXTRACT(a, '$**.analyzing_range_alternatives')) from t200;
+select JSON_LOOSE(JSON_EXTRACT(a, '$**.analyzing_range_alternatives')) from t200;
```
Comment (mariadb-RuchaDeodhar) [inline mysql-test/main/func_json_detail.test:96]: Please add an empty line here. Also, please add the test case in func_json.test and rerecord the results.

**Review by mariadb-RuchaDeodhar — CHANGES_REQUESTED** [review]

File: sql/item_create.cc (hunk 2 above)
Comment (Weijun-H) [reply sql/item_create.cc:5765]: OK, I will create a new PR for it.

**Review by mariadb-RuchaDeodhar — CHANGES_REQUESTED** [review]
The patch looks good now. Thanks ! three more final changes and I can merge it :)
1) Merge all your patches into 1 commit and write the commit message like so:
MDEV-19160: <TITLE>
2) Please make your patch on top of 10.4(only for json_detailed(). For json json_pretty(), keep it as is, that is on top of 10.11 ) 
3) Also, please rerecord all your failing test. Check the tests below. You will find some of them are failing with difference in result. Just rerecord them and add to the commit.

**Comment by Weijun-H** [comment]
> The patch looks good now. Thanks ! three more final changes and I can merge it :)
> 
> 1. Merge all your patches into 1 commit and write the commit message like so:
>    MDEV-19160: <TITLE>
> 2. Please make your patch on top of 10.4(only for json_detailed(). For json json_pretty(), keep it as is, that is on top of 10.11 )
> 3. Also, please rerecord all your failing test. Check the tests below. You will find some of them are failing with difference in result. Just rerecord them and add to the commit. Some compiling also fails with "sql\item_jsonfunc.cc(370,34): warning C4244:         value_len= je->value_end - je->value_begin;" on windows build. Fix them too.

I created a new pr #2410 for rebase the change of `json_detail`
